fedimintd/src/fedimintd.rs

```diff
@@ -257,5 +260,24 @@ async fn run(
         },
         db,
     };
-    api.run(task_group).await
+    let (api_result, metrics_api_result) = futures::join!(
+        api.run(task_group.clone()),
+        spawn_metrics_server(&opts.bind_metrics_api, task_group)
+    );
+    api_result?;
+    metrics_api_result?;
+    Ok(())
+}
+
+async fn spawn_metrics_server(
+    bind_address: &SocketAddr,
+    mut task_group: TaskGroup,
+) -> anyhow::Result<()> {
+    let rx = fedimint_metrics::run_api_server(bind_address, &mut task_group).await?;
```

Comment:
We don't have to spawn anything (other than maybe prom http request handling thread on another port if we want one) and run any periodic metrics update loop. You can see how people usually handle it in some existing projects. I looked for dependents of `prometheus` and `electrs` is one:

Seems like they pass `Metrics` handle to the each major component like Mempool](https://github.com/romanz/electrs/blob/c473556401fd6e56f3411a668a5da655b4298bac/src/mempool.rs#L50), then [update each metric to the right value somewhere next to where the corresponding logic handles it](https://github.com/romanz/electrs/blob/c473556401fd6e56f3411a668a5da655b4298bac/src/mempool.rs#L140).

That's also how I would do it, except maybe I would make all metrics static globals because I'm lazy, and by definition one will have one metrics per the whole process anyway and it's not a library, but an application.

Reply:
This server is to serve the metric results, not to run the metrics update loop